src/lib/state/state.svelte.ts

```diff
@@ -64,11 +67,66 @@ export const globalState = $state<GlobalState>({
     // Indicates if user is using existing approval process, creating one or injected provider
     approvalType: 'existing',
 
+    constructorArguments: {
+      values: {},
+      required: 0
+    },
+
+    deterministic: {
+      isSelected: false,
+      salt: undefined
+    },
+
     // Indicates if deployment is completed.
     completed: false,
   },
 });
 
+export const isValidFormAuthentication = () => globalState.authenticated
+
+const isFormFilledFor = (formInputToCheck: keyof typeof globalState.form) => () => Boolean(globalState.form[formInputToCheck])
+export const isValidFormNetwork = isFormFilledFor("network")
+
+export const isValidConstructorArguments = () => Object.keys(globalState.form.constructorArguments.values).length === globalState.form.constructorArguments.required
+
+export const isValidDeterministicConfiguration = () => globalState.form.deterministic.isSelected ? Boolean(globalState.form.deterministic.salt) : true
+
+export const isValidFormApprovalProcess = async () => {
+  if (globalState.form.approvalType === "injected") {
+    return true;
+  }
+
+  if (globalState.form.approvalType === 'existing' && globalState.form.approvalProcessSelected) {
+    return true;
+  }
+
+  if (
+    globalState.form.approvalType === 'new' &&
+    globalState.form.approvalProcessToCreate?.viaType === "Relayer" &&
+    globalState.form.approvalProcessToCreate?.relayerId
+  ) {
+    return true;
+  }
+
+  if (
+    globalState.form.approvalType === 'new' &&
+    globalState.form.approvalProcessToCreate?.viaType !== "Relayer" &&
+    globalState.form.approvalProcessToCreate?.via
+  ) {
+
+    const checkIfValidAddress = async () =>
+      getAddress(globalState.form.approvalProcessToCreate!.via!)
+    const [_checkSumed, error] = await attempt(checkIfValidAddress);
+    if (error) {
+      return false;
+    }
+    return true;
+  }
+
+  return false;
+}
```

Comment:
Nice refactor :) this was something I wanted to do before but for some reason I forgot it